src/v2/Apps/Auction2/Routes/Bid/useSubmitBid.ts

```diff
@@ -0,0 +1,204 @@
+import { Router } from "found"
+import { useRef } from "react"
+import { useRouter } from "v2/System/Router/useRouter"
+import createLogger from "v2/Utils/logger"
+import { useBidderPosition } from "v2/Apps/Auction2/Queries/useBidderPosition"
+import { useCreateBidderPosition } from "v2/Apps/Auction2/Queries/useCreateBidderPosition"
+import {
+  AuctionFormHelpers,
+  AuctionFormValues,
+  BiddingStatus,
+  errorMessageForBidding,
+} from "v2/Apps/Auction2/Components/Form/Utils"
+import { useCreateTokenAndSubmit } from "v2/Apps/Auction2/Hooks/useCreateTokenAndSubmit"
+import { useAuctionTracking } from "v2/Apps/Auction2/Hooks/useAuctionTracking"
+import { Auction2BidRoute_sale } from "v2/__generated__/Auction2BidRoute_sale.graphql"
+import { Auction2BidRoute_artwork } from "v2/__generated__/Auction2BidRoute_artwork.graphql"
+
+const logger = createLogger("useSubmitBid")
+
+export const useSubmitBid = ({
+  artwork,
+  bidderID,
+  me,
+  requiresPaymentInformation,
+  sale,
+}) => {
+  // FIXME
+  const registrationTracked = useRef(false)
+  const { router } = useRouter()
+  const { tracking } = useAuctionTracking()
+  const { submitMutation: createBidderPosition } = useCreateBidderPosition()
+  const { fetchBidderPosition } = useBidderPosition()
+  const { createToken } = useCreateTokenAndSubmit({ me, sale })
+
+  /**
+   * The submitBid flow is as follows:
+   *
+   *  1. If no CC, create a new token
+   *  2. Create a bidder position
+   *  3. Check bidder position status
+   *  4. If PENDING, continue polling (can be removed with new Bidding engine)
+   */
+  const submitBid = async (
+    values: AuctionFormValues,
+    helpers: AuctionFormHelpers
+  ) => {
+    const { checkBidStatus } = setupCheckBidStatus({
+      helpers,
+      tracking,
+      fetchBidderPosition,
+      bidderID,
+      router,
+      sale,
+      artwork,
+    })
+
+    if (requiresPaymentInformation) {
+      try {
+        await createToken(values, helpers)
+      } catch (error) {
+        logger.error("Error creating token", error)
+      }
+    }
```

Comment:
Since each of these can throw unique errors each needs to be try / caught and in the next PR this will be hooked up to an error handler for directing this stuff back to the UI